output/cloud/v1/output.go

```diff
@@ -101,6 +110,49 @@ func (out *Output) Start() error {
 		}()
 	}
 
+	if insightsOutput.Enabled(out.config) {
+		testRunID, err := strconv.ParseInt(out.referenceID, 10, 64)
+		if err != nil {
+			return err
+		}
+		out.requestMetadatasCollector = insightsOutput.NewCollector(testRunID)
+
+		insightsClientConfig := insights.ClientConfig{
+			IngesterHost: out.config.TracesHost.String,
+			Timeout:      types.NewNullDuration(90*time.Second, false),
+			AuthConfig: insights.ClientAuthConfig{
+				Enabled:                  true,
+				TestRunID:                testRunID,
+				Token:                    out.config.Token.String,
+				RequireTransportSecurity: true,
+			},
+			TLSConfig: insights.ClientTLSConfig{
+				Insecure: false,
+			},
+			RetryConfig: insights.ClientRetryConfig{
+				RetryableStatusCodes: `"UNKNOWN","INTERNAL","UNAVAILABLE","DEADLINE_EXCEEDED"`,
+				MaxAttempts:          3,
+				PerRetryTimeout:      30 * time.Second,
+				BackoffConfig: insights.ClientBackoffConfig{
+					Enabled:        true,
+					JitterFraction: 0.1,
+					WaitBetween:    1 * time.Second,
+				},
+			},
+		}
+		insightsClient := insights.NewClient(insightsClientConfig)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Same comment about the 10 seconds https://github.com/grafana/k6/pull/3201#discussion_r1265632026

Reply:
Even if we agreed that making this configurable isn't a priority, it's still better to keep this along with the other configuration (part of the `ClientConfig`). That way, we will be sure that now the configuration is distributed (:crossed_fingers:), and the next dev who should adjust it thank us since it's pretty clear where it should be adjusted.

Also, do you think about making the connection logic internal thing of the sight client, by the way? I mean, introduce something like:

```golang
func (c *Client) Connect() error {
   ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
   defer cancel()
   if err := insightsClient.Dial(ctx); err != nil {
		return err
	}

    return nil
}
```

Reply:
>Even if we agreed that making this configurable isn't a priority, it's still better to keep this along with the other configuration (part of the ClientConfig).

I completely disagree with this one. I think abstracting away `context.Context` in functions that are indeed blocking is a bad design. Also, the pattern of having a "global default timeout" and cancellation semantics via `context.Context` is well established in Golang - you can take a look at `http.Client` for example.

Reply:
Well, I agree that abstracting away from the context is a bad idea, but basically, the proposal is based on the existing code where you use top-level `context.Background()`. Ideally, it should accept the context, and the signature for the `Connect` should look like `Connect(ctx context.Context)`.

Still, my main point was about encapsulating the client's timeout logic into the package code & consolidating the config. Since duplicating this in the v1 & v2 again outputs a pretty strong signal.

Reply:
>However, I see no value in doing this as there is no use case.

The value here is not having distributed configuration across code. You already have the structure that keeps the config, so why the other configuration thing should be separated?

Reply:
I think I'm lost.

> Well, I agree that abstracting away from the context is a bad idea, but basically, the proposal is based on the existing code where you use top-level context.Background().

Why would this be better than explicitly using a context with a set timeout? If anything, that's worse.

> Ideally, it should accept the context, and the signature for the Connect should look like Connect(ctx context.Context). That way it will respect any another timeout that will come with the context.

Yeah, but again - what problem does it solve? Setting up this in two places is relatively easy. A little copy is better than over-engineering (it will require adding tests, etc.). It's not going to be used in another place. It will soon be removed from the v1 output once it gets deprecated and removed altogether.

Reply:
> Yeah, but again - what problem does it solve? 

Again, it **consolidates** the config and makes things predictable and explicit. 

In your current case, you have to spend some time to learn that line
```
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
is dialing timeout. So having instead of [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) `10*time.Second` is something like `cfg.DialTimeout` fixes explicit issues.

The other is predictable. When I'm looking into the struct like https://github.com/grafana/k6/pull/3202/files#diff-fb2855cb8726c46cb031a9973fb066a19fc0b37cf60faf9a631cee8ab6a5997cR94-R117 I see a lot of configurations there, along with multiply boolean. Why, for instance, are they there?

>A little copy is better than over-engineering (it will require adding tests, etc.). I

I'm not following you on the `over-engineering`. Is adding a one-more field over-engineering? :thinking: Why do you think that it immediately requires the tests? Why then right now we have no tests?

Reply:
It seems to me the discussion drifted off a bit from its starting point to touch on what each of us value and consider bad or good engineering. From my perspective, both positions stand on their feet, and unfortunately, we're not going to solve it here and now (personally, I think both your positions make sense).

As it looks like we reached kind of a deadlock, I would offer that we base our decision on "practices" that the k6 codebase has been following so far to keep it consistent 🤝

A quick [GitHub search](https://github.com/search?q=repo%3Agrafana%2Fk6+context.WithTimeout&type=code) outlined to me that in most Go files (that are not testing files) using the `context.WithTimeout` function, we are setting the timeout from _some_ configuration as opposed to hardcoded values.

I suggest we follow that pattern for now and thus apply @olegbespalov's recommendation and that we take this discussion someplace else if we deem it worth discussing further 🙇🏻

Reply:
Done in [0844bcb](https://github.com/grafana/k6/pull/3202/commits/0844bcb429b091c230a4a14c6ad886148da71e7c).

Reply:
Thanks @Blinkuu, appreciate it 🙇🏻